pages/login.py

```diff
@@ -29,12 +29,15 @@
 #         return {"error": str(e)}
 
 # Function to assign a condition based on the counts
-def assign_condition(condition_counts):
+def assign_condition(condition_counts, condition_with_higher_odds):
     # Find the condition with the minimum count
     min_count = min(condition_counts.values())
     balanced_conditions = [condition for condition, count in condition_counts.items() if count == min_count]
 
-    # Randomly choose a condition from the balanced list
+    if condition_with_higher_odds in balanced_conditions:
+        balanced_conditions.append(condition_with_higher_odds)
```

Comment:
does this actually result in 2x probability for static-chain condition? this implementation might not generalize well if we want to update the weights for more than one condition. Can we actually just change this to simple weighted random sampling where we can set the weight for each condition?